File: src/ShoppingCart.Business/Domain/Product.cs
using System;

namespace ShoppingCart.Business.Domain
{
    public class Product
    {
        public Guid Id { get; }

        public string Title { get; }

        public Price Price { get; }

        public Guid CategoryID { get; }

        public Product(Guid id, string title, Price price, Guid categoryId)
        {
            Id = id;
            Title = title;
            Price = price;
            CategoryID = categoryId;
        }
    }
}

File: spring-boot/src/main/java/com/erdaldalkiran/cart/domain/product/Product.java
package com.erdaldalkiran.cart.domain.product;

import lombok.Getter;

import java.math.BigDecimal;
import java.util.UUID;

@Getter
public class Product {

    private final UUID id;
    private final String title;
    private final BigDecimal price;
    private final UUID categoryID;

    public Product(UUID id, String title, BigDecimal price, UUID categoryID) {
        this.id = id;
        this.title = title;
        this.price = price;
        this.categoryID = categoryID;
    }
}
